technical-reports/color/color-type.md

```diff
@@ -1,22 +1,229 @@
-# The color type
+# Color type
 
-Represents a 24bit RGB or 24+8bit RGBA color in the sRGB color space. The `$type` property MUST be set to the string `color`. The value MUST be a string containing a hex triplet/quartet including the preceding `#` character. To support other color spaces, such as HSL, translation tools SHOULD convert color tokens to the equivalent value as needed.
+## Format
 
-For example, initially the color tokens MAY be defined as such:
+Colors can be represented through various formats. For color tokens, the `$type` property MUST be set to the string `color`. The `$value` property can then be used to specify the details of the color, including color space, alpha settings, and more. The `$value` object contains the following properties:
+
+- `$hex` (required): the hex color to use as the default or a guaranteed fallback
+- `$colorSpace` (optional): An object detailing an alternative color space to be used to interpret the color, if supported
+
+The `$colorSpace` object has the following properties:
+
+- `$name` (required): the name of the color space (either rgb, srgb, hsl, or lch)
+- `$components` (required): the non-alpha pieces of the color, listed as an array of floating-point numbers or integers
```

Comment:
Oh yes this **definitely needs clarification!** Because no, values won’t always be between `0-1` if we follow [CSS Color Module 4](https://www.w3.org/TR/css-color-4/). There are 2 reasons:

1. **Degrees**. As in the case of HSL, the range is 0°–360°. This could probably be normalized to `1` but I think most would agree that fails the goal of this spec being human writable since that’s nonstandard
2. **Device-independent colorspaces.** In the case of colorspaces like LAB/LCH and OKLAB/OKLCH, they describe many colors that are outside the range of any device existing today, and so there’s no “maximum.”

For the latter, CSS Color Module 4 still does define allowable ranges which are still far beyond what most devices can support; I believe they just put reasonable caps on values that decades of hardware improvements still wouldn’t catch up to. Here are some of the values:

- Lab: L [0 … 100], a [-125 … 125], b [-125 … 125]
- Lch: L [0 … 100], c [0 … 150], h [0 … 360]
- Oklab: L [0 … 1], a [-0.4 … 0.4], b [-0.4 … 0.4]
- Oklab: L [0 … 1], c [0 … 0.4], h [0 … 360]
- XYZ: 🤔 for some reason this is normalized [0 … 1] but not sure why

The general approach of “match CSS Color Module 4 as much as possible” has been discussed in meetings but I think it would be good to explicate this as much as possible. As far as I know, the color proposal wouldn’t deviate from Color Module 4 in any way. cc @resource11

Reply:
Also `alpha` in all cases is always normalized [0 … 1], and is optional (default `1` if omitted).

Reply:
@drwpow yeah, if no color space is omitted, and no other color function is used, opacity/alpha channel (rgba) should always default to normalized 0-1